src/pages/Facility/settings/devices/components/DeviceForm.tsx

```diff
@@ -340,6 +340,11 @@ export default function DeviceForm({ facilityId, device, onSuccess }: Props) {
                     type="date"
                     {...field}
                     data-cy="manufacture-date-input"
+                    onChange={(e) => {
+                      field.onChange(e);
+                      // Trigger validation for expiration date when manufacture date changes
+                      form.trigger("expiration_date");
+                    }}
```

Comment:
there's a mode setting in the useForm's options, which can be set to `onChange`. that way we wouldn't need to do this manually.